File: Reservo/Reservo.UI/reservo_organizer/lib/src/models/venue_request/venue_request.dart
import 'package:json_annotation/json_annotation.dart';

part 'venue_request.g.dart';

@JsonSerializable()
class VenueRequest {
  final int id;
  final String? organizerName;
  final String venueName;
  final String cityName;
  final String address;
  final int capacity;
  final String? description;
  final List<int> allowedCategoryIds;
  final String suggestedCategories;
  final String state;
  final DateTime createdAt;


  VenueRequest({
    required this.id,
    this.organizerName,
    required this.venueName,
    required this.cityName,
    required this.address,
    required this.capacity,
    this.description,
    required this.suggestedCategories,
    required this.allowedCategoryIds,
    required this.state,
    required this.createdAt,
  });

  factory VenueRequest.fromJson(Map<String, dynamic> json) => _$VenueRequestFromJson(json);

  Map<String, dynamic> toJson() => _$VenueRequestToJson(this);
}

File: Reservo/Reservo.Model/Entities/VenueRequest.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Reservo.Model.Entities
{
    public class VenueRequest
    {
        public int Id { get; set; }
        [ForeignKey(nameof(User))]
        public int OrganizerId { get; set; }
        public User Organizer { get; set; }
        public string VenueName { get; set; }
        public string CityName { get; set; }
        public string Address { get; set; }
        public int Capacity { get; set; }
        public string? Description { get; set; }
        public ICollection<VenueRequestCategory> VenueRequestCategories { get; set; } = new List<VenueRequestCategory>();
        public string? SuggestedCategories { get; set; }
        public string State { get; set; } = "Pending";
        public DateTime CreatedAt { get; set; } = DateTime.Now;
    }

}
